docs/docs/features/modules.mdx

```diff
@@ -0,0 +1,182 @@
+---
+title: Modules
+sidebar_label: Modules
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+ZMK makes use of [Zephyr modules](https://docs.zephyrproject.org/3.5.0/develop/modules.html) to include additional source code or configuration files into its build. You can think of them as similar to plugins or themes. The most common uses of this feature are:
+
+- Building firmware for a keyboard external to ZMK's tree
+- Adding functionality to ZMK, such as a driver or a behaviour
+
+A common ZMK setup thus consists of the following separate Git repositories:
+
+- A single ZMK config maintained by the user, containing the `.conf` and `.keymap` files for one or multiple keyboards. This is also where files from ZMK or modules should be overridden/modified, if there is a need.
+- Any number of ZMK modules, maintained by the module's owner. Some modules may contain multiple keyboards or functionalities. If all of your keyboards and functionalities are internal to ZMK's tree, then no modules are necessary.
+- The ZMK firmware itself, maintained by its contributors.
+
+:::note
+The shift to using modules for keyboards is a relatively recent one, and not all designs may be properly configured to be used as a module. If this is the case for your keyboard, then we would strongly suggest asking your vendor or designer to rectify this.
+:::
+
+## Building With Modules
+
+Adding modules to your ZMK build is as simple as modifying the `west.yml` found in your `zmk-config`'s `config` directory. The recommended way of doing so is:
```

Comment:
Can we add a subheader here for "GitHub Actions" (with a link to `../user-setup.mdx` in the body) to put this section below it, then another subheader for local builds with a sentence linking to `../development/build-flash.mdx#building-with-external-modules)`?

Reply:
Alternatively, we can mention the first in a sentence in the beginning and add a short sentence at the end for the latter.

Reply:
Added subheaders and sentences.